erpnext/stock/doctype/item/item.py

```diff
@@ -2,744 +2,777 @@
-	make_variant_item_code, validate_item_variant_attributes, ItemVariantExistsError)
+from frappe.website.website_generator import WebsiteGenerator
+
+
+class DuplicateReorderRows(frappe.ValidationError):
+    pass
+
+
+class StockExistsForTemplate(frappe.ValidationError):
+    pass
 
-class DuplicateReorderRows(frappe.ValidationError): pass
-class StockExistsForTemplate(frappe.ValidationError): pass
 
 class Item(WebsiteGenerator):
-	website = frappe._dict(
-		page_title_field = "item_name",
-		condition_field = "show_in_website",
-		template = "templates/generators/item.html",
-		no_cache = 1
-	)
-
-	def onload(self):
-		super(Item, self).onload()
-
-		self.set_onload('stock_exists', self.stock_ledger_created())
-		if self.is_fixed_asset:
-			asset = frappe.db.get_all("Asset", filters={"item_code": self.name, "docstatus": 1}, limit=1)
-			self.set_onload("asset_exists", True if asset else False)
-
-	def autoname(self):
-		if frappe.db.get_default("item_naming_by")=="Naming Series":
-			if self.variant_of:
-				if not self.item_code:
-					template_item_name = frappe.db.get_value("Item", self.variant_of, "item_name")
-					self.item_code = make_variant_item_code(self.variant_of, template_item_name, self)
-			else:
-				from frappe.model.naming import make_autoname
-				self.item_code = make_autoname(self.naming_series+'.#####')
-		elif not self.item_code:
-			msgprint(_("Item Code is mandatory because Item is not automatically numbered"), raise_exception=1)
-
-		self.item_code = strip(self.item_code)
-		self.name = self.item_code
-
-	def before_insert(self):
-		if not self.description:
-			self.description = self.item_name
-
-		if self.is_sales_item and not self.get('is_item_from_hub'):
-			self.publish_in_hub = 1
-
-	def after_insert(self):
-		'''set opening stock and item price'''
-		if self.standard_rate:
-			self.add_price()
-
-		if self.opening_stock:
-			self.set_opening_stock()
-
-	def validate(self):
-		self.get_doc_before_save()
-
-		super(Item, self).validate()
-
-		if not self.item_name:
-			self.item_name = self.item_code
-
-		if not self.description:
-			self.description = self.item_name
-
-		self.validate_uom()
-		self.validate_description()
-		self.add_default_uom_in_conversion_factor_table()
-		self.validate_conversion_factor()
-		self.validate_item_type()
-		self.check_for_active_boms()
-		self.fill_customer_code()
-		self.check_item_tax()
-		self.validate_barcode()
-		self.validate_warehouse_for_reorder()
-		self.update_bom_item_desc()
-		self.synced_with_hub = 0
-
-		self.validate_has_variants()
-		self.validate_stock_exists_for_template_item()
-		self.validate_attributes()
-		self.validate_variant_attributes()
-		self.validate_website_image()
-		self.make_thumbnail()
-		self.validate_fixed_asset()
-		self.validate_retain_sample()
-
-		if not self.get("__islocal"):
-			self.old_item_group = frappe.db.get_value(self.doctype, self.name, "item_group")
-			self.old_website_item_groups = frappe.db.sql_list("""select item_group
```

Comment:
Why spaces over tabs? In standard, we use tabs for indentation. Kindly don't change the format.

Reply:
that's was a pep8 standard thought

Reply:
Oh, we like to use tabs not the spaces!
https://www.youtube.com/watch?v=SsoOG6ZeyUI